### PR TITLE
[SP login] add SP login form

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+    @Override
+    protected @NotNull List<ValidationInfo> doValidateAll() {
+        List<ValidationInfo> res = new ArrayList<>();
+        AuthConfiguration auth = getData();
+        if (StringUtils.isBlank(auth.getTenant())) {
+            res.add(new ValidationInfo("tenant is required.", tenantIdTextField));
+        }
+        if (!isGuid(auth.getTenant())) {
+            res.add(new ValidationInfo("tenant should be a guild.", tenantIdTextField));
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java:136]: typo

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+                } catch (JsonSyntaxException ex) {
+                    // ignore all json errors
+                }
+            }
+        } finally {
+            intermediateState = false;
+        }
+    }
+
+    private static boolean isPlaceHolder(String password) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java:281]: -> Placeholder

**Review by Flanker32 — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java (hunk 1 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java:136]: resolved in https://github.com/microsoft/azure-tools-for-java/pull/5265

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java (hunk 2 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java:281]: see example: 

package com.intellij.ui;

public interface PlaceHolder

**Pull request merged**